### ISSUE TITLE
[Request]: Force Camera App Open

### ISSUE THREAD

**Issue by Rabtos** [Yimura/YimMenu#2174]
### Problem

The camera closes when you are on moving objects, when you get shot or are in any other animation.
Also you can't do a selfie in a vehicle. 

### Solution

Add an option to force open the Camera App.

### Reason

Sometimes you want to take a picture and random people shoot you, drive over you or whatever. ITs really annoying if you often do pictures. Also would be great for some funny and cool pics which you could upload to your Rockstar Socialclub Profil Gallery.

**Comment by pierrelasse** [comment]
Yeah it would be REALLY awesome if you could upload .png or whatever to socialclub.

**Comment by alyxme** [comment]
The camera application can work simultaneously with the noclip function, which in turn eliminates all of the above interruptions. I've taken some cool shots straight out of the camera with noclip feature, without needing to use the Editor. You can check if this is a suitable solution.

**Comment by Rabtos** [comment]
> Yeah it would be REALLY awesome if you could upload .png or whatever to socialclub.

Dat's not really what I mean. I guess this has nothing to do with YimMenu?

> The camera application can work simultaneously with the noclip function, which in turn eliminates all of the above interruptions. I've taken some cool shots straight out of the camera with noclip feature, without needing to use the Editor. You can check if this is a suitable solution.

Yeah I know that, I'M also using Freecamera Mode. BUT doing a pic while standing on a Jet or a Boat wouldn't work with NoClip,  you would just clip through.. so dat's not the suitable solution I am looking for... But Thanks  👍

**Comment by Rabtos** [comment]
I found this: 

```
// 0xFDE8F069C542D126 0x234C1AE9 b323
void CELL_​CAM_​ACTIVATE(
BOOL p0,
BOOL p1
)
```

Script usage

```
// appcamera.ysc @ L9283
void func_108(int iParam0, int iParam1)
{
  if (Global_4456448.f_185406 == 1)
  {
  }
  else if (Global_4456448.f_185407 == 1)
  {
  }
  else
  {
    MOBILE::CELL_CAM_ACTIVATE(iParam0, iParam1);
  }
}
```

https://nativedb.dotindustries.dev/natives/0xFDE8F069C542D126?search=mobile
https://nativedb.dotindustries.dev/natives/0x015C49A93E3E086E?search=mobile
https://www.lcpdfr.com/resources/nativedb/index/mobile/

Is there any way to make an .lua to activate this function? 
Or any other way to make something out of this?

And MenyooSP has this:
![grafik](https://github.com/YimMenu/YimMenu/assets/124796345/72aa40ad-afb2-43f1-86fc-5bfd8912ec07)
https://github.com/MAFINS/MenyooSP/blob/master/Solution/source/Natives/natives.h

Can someone explain what this is? I guess those "natives" are something like make those functions accesible by lua scripts?

**Comment by gir489returns** [comment]
@Rabtos use MOBILE.CELL_CAM_ACTIVATE. We already have these natives in use for the Lua runner.

https://github.com/YimMenu/YimMenu/blob/ba5610af99da7e450d497ad7c9decdad57880fad/src/lua/natives/lua_native_binding_MOBILE.cpp#L151

**Comment by Rabtos** [comment]
@gir489returns 

Thanks for your help! So looks like it's possible to use YIMs Lua to make this feature.
But I'm a noob and don't really understand how to implement things.. I tried this code and it does nothing:

```
cellcamTab = gui.get_tab("GUI_TAB_FUN_VEHICLE")

script.register_looped("ActivateSelfieCam", function()
    if ActivateSelfieCam:is_enabled() then
		MOBILE.set_function("CELL_CAM_ACTIVATE", LUA_NATIVE_MOBILE_CELL_CAM_ACTIVATE);
	end
end)

ActivateSelfieCam = cellcamTab:add_checkbox("Toggle Cellphone Camera")
```

**Comment by gir489returns** [comment]
@Rabtos 

You don't understand how the scripts, I suggest you read the documentation more thoroughly. The natives are meant to be called just like they are in the scripts but in Lua. The code I showed you was only to prove the C++ functions are being exposed in Lua. In RAGE you would call it like this

```
// appcamera.ysc @ L9283
void func_108(int iParam0, int iParam1)
{
  if (Global_4456448.f_185406 == 1)
  {
  }
  else if (Global_4456448.f_185407 == 1)
  {
  }
  else
  {
    MOBILE::CELL_CAM_ACTIVATE(iParam0, iParam1);
  }
}
```

The only difference is, Lua doesn't have namespaces, everything belongs to the global table, it only has the subtable operator which is the period.

So converted to Lua, it would look like this:

`MOBILE.CELL_CAM_ACTIVATE(iParam0, iParam1);`

The semi-colon operator is a comment operator in Lua, so it doesn't matter if you leave it in or not. 

param0 and param1 are both Booleans, but it is unknown what their function is. In the scripts, they are generally called with false, though.
